assets/__tests__/live_socket_test.js

```diff
@@ -0,0 +1,219 @@
+
+  expect(liveSocket.getViewById('container').destroy).toBeDefined();
+});
+
+test('destroyViewById', async () => {
+  let liveSocket = new LiveSocket('/live');
+  liveSocket.document = dom().window.document;
+
+  liveSocket.connect();
+
+  liveSocket.destroyViewById('container');
+  expect(liveSocket.getViewById('container')).toBeUndefined();
+});
+
+test('getBindingPrefix', async () => {
+  let liveSocket = new LiveSocket('/live');
+
+  expect(liveSocket.getBindingPrefix()).toEqual('phx-');
+});
+
+test('getBindingPrefix custom', async () => {
+  let liveSocket = new LiveSocket('/live', { bindingPrefix: 'company-' });
+
+  expect(liveSocket.getBindingPrefix()).toEqual('company-');
+});
+
+test('getActiveElement default before LiveSocket activeElement is set', async () => {
+  let liveSocket = new LiveSocket('/live');
+  liveSocket.document = dom().window.document;
+
+  let input = liveSocket.document.querySelector('input');
+  input.focus();
+
+  expect(liveSocket.getActiveElement()).toEqual(input);
+});
+
+test('setActiveElement and getActiveElement', async () => {
+  let liveSocket = new LiveSocket('/live');
+  liveSocket.document = dom().window.document;
+
+  let input = liveSocket.document.querySelector('input');
+
+  // .activeElement
+  liveSocket.setActiveElement(input);
+  expect(liveSocket.activeElement).toEqual(input);
+  expect(liveSocket.getActiveElement()).toEqual(input);
+});
+
+test('blurActiveElement', async () => {
+  let liveSocket = new LiveSocket('/live');
+  liveSocket.document = dom().window.document;
+
+  let input = liveSocket.document.querySelector('input');
+  input.focus();
+
+  expect(liveSocket.prevActive).toBeNull();
+
+  liveSocket.blurActiveElement();
+  // sets prevActive
+  expect(liveSocket.prevActive).toEqual(input);
+  expect(liveSocket.getActiveElement()).not.toEqual(input);
+});
+
+test('restorePreviouslyActiveFocus', async () => {
+  let liveSocket = new LiveSocket('/live');
+  liveSocket.document = dom().window.document;
+
+  let input = liveSocket.document.querySelector('input');
+  input.focus();
+
+  liveSocket.blurActiveElement();
+  expect(liveSocket.prevActive).toEqual(input);
+  expect(liveSocket.getActiveElement()).not.toEqual(input);
+
+  // focus()
+  liveSocket.restorePreviouslyActiveFocus();
+  expect(liveSocket.prevActive).toEqual(input);
+  expect(liveSocket.getActiveElement()).toEqual(input);
+});
+
+test('dropActiveElement unsets prevActive', async () => {
+  let liveSocket = new LiveSocket('/live');
+  liveSocket.document = dom().window.document;
+
+  liveSocket.connect();
+
+  let input = liveSocket.document.querySelector('input');
+  liveSocket.setActiveElement(input);
+  liveSocket.blurActiveElement();
+  expect(liveSocket.prevActive).toEqual(input);
+
+  let view = liveSocket.getViewById('container');
+  liveSocket.dropActiveElement(view);
+  expect(liveSocket.prevActive).toBeNull();
+  // this fails.  Is this correct?
+  // expect(liveSocket.getActiveElement()).not.toEqual(input);
```

Comment:
@chrismccord This was the only question I had when doing the tests.  So `dropActiveElement` still leaves the `activeElement` as the `input`.  I would have expected it to also drop `this.activeElement` && / or `document.activeElement`.  Any thoughts?

Reply:
This is still applicable.  I thought the active element should be empty...but still is an `input`.